File: Source/CSharp/Expressive/Expressive/Functions/ExpFunction.cs
using Expressive.Expressions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Expressive.Functions
{
    internal class ExpFunction : FunctionBase
    {
        #region FunctionBase Members

        public override string Name { get { return "Exp"; } }

        public override object Evaluate(IExpression[] participants)
        {
            this.ValidateParameterCount(participants, 1, 1);

            return Math.Exp(Convert.ToDouble(participants[0].Evaluate(Arguments)));
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/functions/mathematical/ExpFunction.java
package com.bijington.expressive.functions.mathematical;

import com.bijington.expressive.exceptions.ExpressiveException;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.functions.FunctionBase;

/**
 * Created by shaun on 18/07/2016.
 */
public class ExpFunction extends FunctionBase {
    @Override
    public String getName() {
        return "Exp";
    }

    @Override
    public Object evaluate(IExpression[] parameters) throws ExpressiveException {
        this.validateParameterCount(parameters, 1, 1);

        Object value = parameters[0].evaluate(this.getVariables());

        if (value == null) return null;

        if (value.getClass().equals(Integer.class)) {
            return Math.exp(Integer.class.cast(value));
        }
        else if (value.getClass().equals(Long.class)) {
            return Math.exp(Long.class.cast(value));
        }
        else if (value.getClass().equals(Double.class)) {
            return Math.exp(Double.class.cast(value));
        }
        else if (value.getClass().equals(Float.class)) {
            return Math.exp(Float.class.cast(value));
        }

        return null;
    }
}
